src/dvs2tensor.cpp

```diff
@@ -0,0 +1,296 @@
+
+    // Let's verify they really changed!
+    uint32_t prBias   = davisHandle.configGet(DAVIS_CONFIG_BIAS, DAVIS240_CONFIG_BIAS_PRBP);
+    uint32_t prsfBias = davisHandle.configGet(DAVIS_CONFIG_BIAS, DAVIS240_CONFIG_BIAS_PRSFBP);
+
+    printf("New bias values --- PR-coarse: %d, PR-fine: %d, PRSF-coarse: %d, PRSF-fine: %d.\n",
+    caerBiasCoarseFineParse(prBias).coarseValue, caerBiasCoarseFineParse(prBias).fineValue,
+    caerBiasCoarseFineParse(prsfBias).coarseValue, caerBiasCoarseFineParse(prsfBias).fineValue);
+
+}
+
+// Now let's get start getting some data from the device. We just loop in blocking mode,
+// no notification needed regarding new events. The shutdown notification, for example if
+// the device is disconnected, should be listened to.
+void DVSDataConv::startdatastream(){
+
+    // Let's get configs about buffer
+    davisHandle.configSet(CAER_HOST_CONFIG_DATAEXCHANGE, CAER_HOST_CONFIG_DATAEXCHANGE_BUFFER_SIZE, buffer_size);
+
+    uint32_t BFSize   = davisHandle.configGet(CAER_HOST_CONFIG_DATAEXCHANGE, CAER_HOST_CONFIG_DATAEXCHANGE_BUFFER_SIZE);
+
+    printf("Buffer size: %d", BFSize);
+
+    davisHandle.dataStart(nullptr, nullptr, nullptr, &DVSDataConv::usbShutdownHandler, nullptr);
+
+    // Let's turn on blocking data-get mode to avoid wasting resources.
+    davisHandle.configSet(CAER_HOST_CONFIG_DATAEXCHANGE, CAER_HOST_CONFIG_DATAEXCHANGE_BLOCKING, true);
+ 
+}
+
+
+// Process an event and convert any given PolarityEvents to sparse tensor structure
+torch::Tensor DVSDataConv::update(){
+    std::vector<DVSevents::PolarityEvent> polarity_events;
+    std::unique_ptr<libcaer::events::EventPacketContainer> packetContainer = nullptr;
+
+    do {
+      packetContainer = davisHandle.dataGet();
+    } while (packetContainer == nullptr);
+
+
+    printf("\nGot event container with %d packets (allocated).\n", packetContainer->size());
+
+    for (auto &packet : *packetContainer) {
+        if (packet == nullptr) {
+            printf("Packet is empty (not present).\n");
+            continue; // Skip if nothing there.
+        }
+
+        printf("Packet of type %d -> %d events, %d capacity.\n", packet->getEventType(), packet->getEventNumber(), packet->getEventCapacity());
+
+        if (packet->getEventType() == POLARITY_EVENT) {
+            std::shared_ptr<const libcaer::events::PolarityEventPacket> polarity = std::static_pointer_cast<libcaer::events::PolarityEventPacket>(packet);
+            // Print out timestamps and addresses.
+
+            printf("Lowest Timestamp: %f\n", (float (packetContainer->getLowestEventTimestamp())/1000000));
+            printf("Highest Timestamp: %f\n", (float (packetContainer->getHighestEventTimestamp())/1000000));
+            printf("Time span of polarity packet: %ld\n", (packetContainer->getHighestEventTimestamp() - packetContainer->getLowestEventTimestamp()));
```

Comment:
printing, alternatives: collect statistics somewhere, provide a function to print those statistics.